### PR TITLE
68 purging tests

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

Enable and refactored purging tests on angular 10

medic/angular10-migration [#68](https://github.com/medic/angular10-migration/issues/68)

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/login/purge.spec.js [hunk 1]
```diff
@@ -248,7 +247,7 @@ describe('Purging on login', () => {
 
   const restartSentinel = () => utils.stopSentinel().then(() => utils.startSentinel());
 
-  it('Logging in as a restricted user with configured purge rules should not download purged docs', () => {
+  it('Logging in as a restricted user with configured purge rules should not download purged docs', async () => {
```
Comment (newtewt) [inline tests/e2e/login/purge.spec.js:250]: As a general statement, I think adding await here is a bit confusing without reworking the entire promise chaining.  This sample code may be a little off as I'm writing it in the browser. Let me know if this needs a bit more clarity and we can jump on a call. 

```
let result = await getPurgeLog();
// purge ran but after initial replication, nothing to purge
chai.expect(result._rev).to.equal('0-1');
chai.expect(result.roles).to.equal(JSON.stringify(restrictedUser.roles.sort()));
chai.expect(result.history.length).to.equal(1);
chai.expect(result.count).to.equal(0);
chai.expect(result.history[0]).to.deep.equal({
          count: 0,
          roles: result.roles,
          date: result.date
        });
purgeDate = result.date;

await utils.resetBrowser();
await commonElements.calm();
await broweser.waitForAngular();
result = await getPurgeLog();
```

Comment (ngaruko) [reply tests/e2e/login/purge.spec.js:250]: fiexd

File: tests/e2e/login/purge.spec.js [hunk 2]
```diff
@@ -248,86 +247,76 @@ describe('Purging on login', () => {
...
-        // purge didn't run again on next refresh
-        chai.expect(result._rev).to.equal('0-1');
-        chai.expect(result.date).to.equal(purgeDate);
-      })
-      .then(() => {
-        browser.wait(() => utils.saveDocs(subsequentReports).then(() => true));
-        commonElements.sync();
-        commonElements.goToReports();
-        reports.expectReportsToExist([goodFormId, goodFormId2, badFormId2]);
+    await browser.wait(async() => {
```
Comment (newtewt) [inline tests/e2e/login/purge.spec.js:286]: `browser.wait` is similar to the `await` command in that it waits for the promise to resolve before moving on. There are two options here I think. Revert to using the existing promise chaining and` browser.wait` or remove the promise chaining and switch it all to use `await`.

**Review by newtewt — CHANGES_REQUESTED** [review]
Nice job reworking that large chain! I think there is a little more to update still.

File: tests/conf.js [hunk 3]
```diff
@@ -30,8 +30,11 @@ const baseConfig = {
       '**/send-message.js',
       '**/message_duplicates.spec.js',
       '**/api/server.js',
-      'e2e/reports-subject.js'
+      'tests/e2e/create-meta-db.js',
```
Comment (newtewt) [inline tests/conf.js:33]: `create-meta-db.js` was moved to `conf-disabled` this can be removed.

**Review by newtewt — APPROVED** [review]
Just need to update 1 thing in the conf file. Otherwise nice work on that large refactor.

**Comment by newtewt** [comment]
Moved to https://github.com/medic/cht-core/pull/6905

**Pull request closed without merging**